### PR TITLE
Use a sentinel value in the lightbulb code to simplify logic.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Host/Status/NoOpWorkspaceStatusService.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+        event EventHandler IWorkspaceStatusService.StatusChanged
+        {
+            add { }
+            remove { }
+        }
+
+        public Task WaitUntilFullyLoadedAsync(CancellationToken cancellationToken)
+        {
+            // by the default, we are always fully loaded
```
Comment (allisonchou) [inline src/Workspaces/Core/Portable/Workspace/Host/Status/NoOpWorkspaceStatusService.cs:28]:
```suggestion
            // by default, we are always fully loaded
```

File: src/Workspaces/Core/Portable/Workspace/Host/Status/NoOpWorkspaceStatusService.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            return Task.CompletedTask;
+        }
+
+        public Task<bool> IsFullyLoadedAsync(CancellationToken cancellationToken)
+        {
+            // by the default, we are always fully loaded
```
Comment (allisonchou) [inline src/Workspaces/Core/Portable/Workspace/Host/Status/NoOpWorkspaceStatusService.cs:34]:
```suggestion
            // by default, we are always fully loaded
```

**Review by allisonchou — APPROVED** [review]
Just some super small comment nits, feel free not to take them 😄

**Review by sharwell — CHANGES_REQUESTED** [review]
Export a default instance of `IWorkspaceStatusService` instead of using a singleton.

**Comment by sharwell** [comment]
I believe we should take https://github.com/dotnet/roslyn/pull/49051 instead of this change.

**Comment by CyrusNajmabadi** [comment]
> Edit: a default instance of WorkspaceStatusService is already available.

Right. But that only works if we have a workspace. When we don't, there's no sentinel that we can use. 

That said, we found change our logic to never store this service and instead always grab it from the workspace when we attach/detach from that.

**Comment by CyrusNajmabadi** [comment]
@sharwell removed the NoOp service in favor of just on demand retrieval of the the status service.

**Review by sharwell — APPROVED** [review]

**Review by sharwell — APPROVED** [review]

**Pull request merged**